HLA-1520: Remove Help Files (2 of 3)

s-goldman opened the pull request:
A continuation of PR #2077. 

This PR moves the IRAF style help files to python docstrings for the following modules:

- refimagefindpars
- pixtopix
- mapreg
- imagefindpars
- drizCR

The documentation that was previously in imagefindpars and refimagefindpars (which are nearly identical) are now apart of the ImageCatalog Class docs in the catalogs.py module. 

Regression test: [Link](https://github.com/spacetelescope/RegressionTests/actions/runs/18565599621)

pllim commented on drizzlepac/mapreg.py:
```diff
@@ -51,6 +62,226 @@ def wcs_pix2sky(self, *args, **kwargs):
+        containing a list of filenames for all input region files with one
+        filename on each line of the file.
+
+        images : string or list of strings (Default = ``*.fits``)
```
Is the extra indent for `images` intentional?

pllim commented on drizzlepac/mapreg.py:
```diff
@@ -51,6 +62,226 @@ def wcs_pix2sky(self, *args, **kwargs):
+    This will produce six region files in the ./regions subdirectory for *each*
+    input image::
+
+        ``img1_sci1_twreg.reg``,    ``img1_sci2_twreg.reg``,    ``img1_sci3_twreg.reg``,
```
You don't need double backticks if the text is already in a literal block (`::`). Looks weird when you use both, please see https://drizzlepac--2078.org.readthedocs.build/en/2078/api/drizzlepac.mapreg.MapReg.html#drizzlepac.mapreg.MapReg

pllim commented on drizzlepac/mapreg.py:
```diff
@@ -125,6 +356,85 @@ def map_region_files(input_reg, images, img_wcs_ext='sci',
+    None
+        All results are persisted to ``outpath`` and optionally to the catalog
+        file.
+
```
No need to say "return None" but you will have to move the note about all results elsewhere, maybe the top.

```suggestion
```

s-goldman commented:
@pllim I've addressed your comments. I also removed an `interactive` argument for mapreg that never has or will be used. In some places it is `iteractive`.

pllim commented on drizzlepac/mapreg.py:
```diff
@@ -49,18 +60,226 @@ def wcs_pix2sky(self, *args, **kwargs):
+            The ``'precise'`` method is not
+            implemented in this release and, if specified, defaults to 'fast'.
+            The ``'precise'`` option is not available through the ``TEAL``
+            interface.
```
Same comment as `refimg`.

```suggestion
```

mcara commented on drizzlepac/mapreg.py:
```diff
@@ -124,7 +343,78 @@ def _simple_parse_teal_extn(extnstr):
 def map_region_files(input_reg, images, img_wcs_ext='sci',
                      refimg=None, ref_wcs_ext='sci', chip_reg=None,
                      outpath='./regions', filter=None, catfname=None,
-                     iteractive=False, append=False, verbose=True):
```
I need to look into this.

> The argument does nothing and has a default of False. If a user did set it as True, then their code would probably raise an error.

This is a contradiction. If the argument "does nothing" then setting it to True should **not** result in an exception. If this argument is no longer needed, deprecate it (adding a changelog entry is not enough). Or ass `**kwargs` or something that is backwards compatible.

mcara commented on drizzlepac/mapreg.py:
```diff
@@ -47,20 +59,226 @@ def wcs_pix2sky(self, *args, **kwargs):
+        Specify whether or not to append the transformed regions to the existing
+        region files with the same name.
+
+    interactive : bool (Default = False)
```
That is, I do not recommend https://github.com/spacetelescope/drizzlepac/pull/2078#discussion_r2467533931 change

pllim commented on CHANGELOG.rst:
```diff
@@ -17,10 +17,11 @@ number of the code change for that issue.  These PRs can be viewed at:
 
-- Removed help files and moved text to docstrings. [#2077,#2078,#2079]
+- Removed help files and moved text to docstrings. Also deprecated the 
+  ``interactive`` argument in ``MapReg`` function. [#2077,#2078,#2079]
```
```suggestion
  ``iteractive`` argument in ``MapReg`` function. [#2077,#2078,#2079]
```
